src/lib/renderComponent.tsx

```diff
@@ -76,6 +77,9 @@ const renderComponent = <P extends {}>(
         const ElementType = getElementType({ defaultProps }, props)
         const rest = getUnhandledProps({ handledProps }, props)
 
+        const accessibility = getAccessibility(props, state)
+        addKeyDownHandler(rest, actions, accessibility, props)
```

Comment:
Since rest/props are component centric ideas, proposing we merge these in the addKeyDownHandler signature so that the method is not aware of internal component implementations.

`addKeyDownHandler({ ...rest, ...props}, actions, accessibility)`

Now, there is just a `props` argument if you will.